File: obstaculos.py
import cv2

# Suponiendo que 'frame' es un cuadro obtenido de una cámara o video
obstacle = False
gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
_, thresh = cv2.threshold(gray, 127, 255, cv2.THRESH_BINARY)
contours, _ = cv2.findContours(thresh, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)

for contour in contours:
    if cv2.contourArea(contour) > 1000:
        obstacle = True
        break

File: obstaculos.cpp
#include <opencv2/opencv.hpp>

int main() {
    cv::Mat frame; // Suponer que 'frame' es obtenido de una cámara
    cv::Mat gray, thresh;
    std::vector<std::vector<cv::Point>> contours;

    cv::cvtColor(frame, gray, cv::COLOR_BGR2GRAY);
    cv::threshold(gray, thresh, 127, 255, cv::THRESH_BINARY);
    cv::findContours(thresh, contours, cv::RETR_TREE, cv::CHAIN_APPROX_SIMPLE);

    for (auto &contour : contours) {
        if (cv::contourArea(contour) > 1000) {
            // Obstáculo detectado
            break;
        }
    }
    return 0;
}
